DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h

```diff
@@ -47,54 +49,84 @@ class MCCompLabel
   // mask for all used fields
   static constexpr ULong64_t maskFull = (ul0x1 << (nbitsTrackID + nbitsEvID + nbitsSrcID)) - 1;
 
-  MCCompLabel(int trackID, int evID = 0, int srcID = 0) { set(trackID, evID, srcID); }
-  MCCompLabel() = default;
+  MCCompLabel(int trackID, int evID, int srcID, bool fake) { set(trackID, evID, srcID, fake); }
+  MCCompLabel(bool noise = false)
+  {
+    if (noise) {
+      mLabel = Noise;
+    } else {
+      mLabel = NotSet;
+    }
+  }
   ~MCCompLabel() = default;
 
   // check if label was assigned
   bool isSet() const { return mLabel != NotSet; }
   // check if label was not assigned
   bool isEmpty() const { return mLabel == NotSet; }
-  // check if label was assigned with non-negaive trackID
-  bool isPosTrackID() const { return getTrackID() >= 0; }
-  // return 1 if the tracks are the same and have labels>=0,
-  // 0 if the tracks are the same but at least one of them has label<0
+  // check if label corresponds to real particle
+  bool isNoise() const { return mLabel == Noise; }
+  // check if label was assigned as for correctly identified particle
+  bool isValid() const { return isSet() && !isNoise(); }
+
+  // check if label was assigned as for incorrectly identified particle or not set or noise
+  bool isFake() const { return (mLabel & Fake) || !isValid(); }
+  // check if label was assigned as for correctly identified particle
+  bool isCorrect() const { return !isFake() && isValid(); }
```

Comment:
You can simplify this slightly, isFake() does not need "&& !isValid()", since NotSet and Noise will both set the Fake bit.
For isCorrect, you can simply use !isFake().